lib/oli/authoring/publishing.ex

```diff
@@ -0,0 +1,27 @@
+defmodule Oli.Authoring.Publishing do
+  import Ecto.Query, warn: false
+
+  alias Oli.Delivery.Sections.Section
+  alias Oli.Repo
+
+  def find_instructors_enrolled_in(section) do
+    lookup_instructors_enrolled_in(section)
+    |> Repo.all()
+  end
+
+  def find_oldest_enrolled_instructor(section) do
+    from(_ in lookup_instructors_enrolled_in(section), limit: 1)
+    |> Repo.one()
+  end
+
+  defp lookup_instructors_enrolled_in(section) do
+    from(s in Section,
+      join: e in assoc(s, :enrollments),
+      join: u in assoc(e, :user),
+      where: s.id == ^section.id,
+      where: u.can_create_sections == true,
+      order_by: [asc: e.inserted_at],
+      select: u
+    )
+  end
```

Comment:
Maybe we could use some of the functions that already exist in `Oli.Delivery.Sections` to get the instructors of a section. 

See if they work for you: 

- **fetch_instructors/1**
- **instructors_per_section/1**
- **get_instructors_for_section/1**